odoo/addons/base/models/ir_actions.py

```diff
@@ -642,7 +762,43 @@ def _run_action_object_write(self, eval_context=None):
             for field, new_value in res.items():
                 record_cached[field] = new_value
         else:
-            self.env[self.model_id.model].browse(self._context.get('active_id')).write(res)
+            starting_record = self.env[self.model_id.model].browse(self._context.get('active_id'))
+            _, _, target_records = self._traverse_path(record=starting_record)
+            target_records.write(res)
+
+    def _run_action_webhook(self, eval_context=None):
+        """Send a post request with a read of the selected field on active_id."""
+        record = self.env[self.model_id.model].browse(self._context.get('active_id'))
+        url = self.webhook_url
+        if not url:
+            raise UserError(_("I'll be happy to send a webhook for you, but you really need to give me a URL to reach out to..."))
+        vals = {
+            '_model': self.model_id.model,
+            '_id': record.id,
+            '_action': '%s(#%s)' % (self.name, self.id),
+        }
+        if self.webhook_field_ids:
+            # you might think we could use the default json serializer of the requests library
+            # but it will fail on many fields, e.g. datetime, date or binary
+            # so we use the json.dumps serializer instead with the str() function as default
+            vals.update(record.read(self.webhook_field_ids.mapped('name'), load=None)[0])
+        json_values = json.dumps(vals, sort_keys=True, default=str)
```

Comment:
> What is the point of sorting the payload keys?
easier to debug + making it similar to the sample payload computed above.

> What is the point of sorting the payload keys? And non-attachment binary fields are going to be all sorts of useless
I guess you meant attachment binary fields would be useless (e.g. those that will send the id of the attachment instead of the actual base64 data). Yeah, i'll see if i can process binary differently depending on those cases